pvl_hsu_soiling.m

```diff
@@ -0,0 +1,351 @@
+PMConcentration=nan(numel(TimeAsDatenum),2); % pre-allocate with NAN
+
+PMConcentration(:,1) = PM2_5; % fill PM2.5 data in column 1
+% Since PM2.5 particulate is counted in PM10 measurements, and we need only
+% the particular from 2.5 micron to 10 micron, subtract the 2.5 micron
+% particulate from the 10 micron particulate
+PMConcentration(:,2) = PM10 - PM2_5; % fill in PM2.5-PM10 data in column 2
+
+% For any instances where 2.5 micron particulate was more thatn 10 micron,
+% set the amount of 2.5 to 10 micron particulate to 0
+PMConcentration(PM10 - PM2_5 < 0 , 2) = 0; 
+
+% EPA concentrations reported in micrograms/m^3, we need it in g/m^3
+PMConcentration = PMConcentration .* 1e-6;
+
+% Calculate the mass deposition rate of each particulate size
+F=PMConcentration .* vd; % g * m^-2 * s^-1, by particulate size
+HorizontalTotalMassRate = F(:,1) + F(:,2); % g * m^-2 * s^-1, total
+
+% Mass depositition rate on a tilted surface. In the event of a changing
+% tilt (e.g. tracking system) the mass rate is adjusted by the tilt angle
+% at each time step, and thus a linear interpolation of changing tilt
+% angles are used (when using trapezoidal integration).
+TiltedMassRate = HorizontalTotalMassRate .* cosd(Tilt);
+
+% Mass on the tilted surface if no rain occurs. Uses trapezoidal
+% integration and determines the amount of mass deposited on a tilted
+% module.
+TiltedMassNoRain = cumtrapz(TimeAsDatenum*86400, TiltedMassRate);
+
+% Create a Tilted soiling mass that will account for rain cleaning
+TiltedMass = TiltedMassNoRain;
+% For each rain accumulation period, check to see if enough rain fell to meet or
+% exceed the cleaning threshold. If the threshold was met or exceeded,
+% then subtract from each subsuequent time step the mass of future time
+% steps. This essentially "resets" the cumulative soiling accumulation when
+% the rain meets or exceeds the stated thresshold. Note that this means
+% that the mass will never be reported as zero because the cleaning is
+% applied immediately after a time step, and new mass accumulates between
+% the cleaning and the first time reported after the cleaning.
+% Note that this could be done more clearly by going through each time step
+% and accumulating or clearing based on daily rainfall, but this is much
+% faster, and can utilize trapezoidal soiling integration.
+for cntr1 = 1:numel(RainAtAccTimes)-1
+    if RainAtAccTimes(cntr1) >= RainThresh
+        TiltedMass(UnqRainAccFrstVal(cntr1+1):end) = TiltedMass(UnqRainAccFrstVal(cntr1+1):end)-TiltedMass(UnqRainAccFrstVal(cntr1+1)-1);
+    end
+end
+
+% Soiling rate is the % transmittance reduction using Heagzy equation
+SoilingRate = 34.37 * erf(0.17 .* (TiltedMass.^0.8473));
+
+% Soiling Ratio is the % soil transmittance (0.95 = 5% soiling loss)
+SR = (100 - SoilingRate) ./ 100;
+
+end
+
+% This function creates deposition velocities for the variable deposition
+% model.
+function vd = pvl_depo_veloc(T, WindSpeed, LUC)
```

Comment:
Minor style point - I'd rather not use the prefix "pvl" for private functions.

Reply:
Corrected, new name is "depo_veloc".